gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/ConnectionPool.java

```diff
@@ -579,6 +614,19 @@ private void announceAllAvailableConnection() {
         }
     }
 
+    private String getAdditionalTimeoutInformation() {
+        final ConnectionResult res = latestConnectionResult.get();
```

Comment:
I have some concerns with the reliance on latestConnectionResult to get the cause of the current timeout. If there are multiple connections being opened and failing at once, couldn't this sometimes get the wrong result for a specific connection?